### PR TITLE
Moves flick_overlay to atom instead of being global

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Moves flick_overlay and flick_overlay_view to atom instead of being a global proc
## Why It's Good For The Game
General performance and syntaxical improvements, makes it easier to retrieve iconstates
## Changelog
:cl:
fix: flick_overlay is an atom proc
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Tests right now only fail for everything that's reading it from global. A cool trick for making sure you fixed everything, including current atom consumers, is to rename the proc `flick_overlay_awesome` or whatever so that all existing usages are a compile error. Once those are fixed, you can rename it back.

File: code/game/objects/items/devices/laserpointer.dm [hunk 1]
```diff
@@ -190,7 +190,7 @@
 			to_chat(user, span_warning("[src]'s battery is overused, it needs time to recharge!"))
 			recharge_locked = TRUE
 
-	flick_overlay_view(I, targloc, 10)
+	I.flick_overlay_view(targloc, 10)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/laserpointer.dm:193]: `I` here is an image and not an atom and so this shouldn't work

**Comment by etherware-novice** [comment]
@Mothblocks also can you rerun the failed checks they went off for no reason

**Comment by Mothblocks** [comment]
They're not invalid--windows build is reporting an unused variable, and lints are giving these errors

image

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -122,16 +122,27 @@
...
 
-///wrapper for flick_overlay(), flicks to everyone who can see the target atom
-/proc/flick_overlay_view(image/image_to_show, atom/target, duration)
+/// Atom-wrapper for flick_overlay_global that converts an iconstate
+/atom/proc/flick_overlay(image_to_show, list/show_to, duration, layer)
+
+	var/image/passed_image = \
+		istext(image_to_show) \
+		? image(icon, src, image_to_show, layer) \
+		: image_to_show
```
Comment (Mothblocks) [inline code/__HELPERS/game.dm:138]:
```suggestion
			? image(icon, src, image_to_show, layer) \
			: image_to_show
```

Bump one to indicate expression scope

File: code/__HELPERS/game.dm [hunk 3]
```diff
@@ -122,16 +122,27 @@
...
+/proc/flick_overlay_global(image/image_to_show, list/show_to, duration)
 	for(var/client/add_to in show_to)
 		add_to.images += image_to_show
 	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(remove_images_from_clients), image_to_show, show_to), duration, TIMER_CLIENT_TIME)
 
-///wrapper for flick_overlay(), flicks to everyone who can see the target atom
-/proc/flick_overlay_view(image/image_to_show, atom/target, duration)
+/// Atom-wrapper for flick_overlay_global that converts an iconstate
+/atom/proc/flick_overlay(image_to_show, list/show_to, duration, layer)
+
```
Comment (Mothblocks) [inline code/__HELPERS/game.dm:134]:
```suggestion
```

Shouldn't space between empty line

File: code/__HELPERS/game.dm [hunk 4]
```diff
@@ -122,16 +122,27 @@
...
 ///Add an image to a list of clients and calls a proc to remove it after a duration
-/proc/flick_overlay(image/image_to_show, list/show_to, duration)
+/proc/flick_overlay_global(image/image_to_show, list/show_to, duration)
 	for(var/client/add_to in show_to)
 		add_to.images += image_to_show
 	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(remove_images_from_clients), image_to_show, show_to), duration, TIMER_CLIENT_TIME)
 
-///wrapper for flick_overlay(), flicks to everyone who can see the target atom
-/proc/flick_overlay_view(image/image_to_show, atom/target, duration)
+/// Atom-wrapper for flick_overlay_global that converts an iconstate
```
Comment (Mothblocks) [inline code/__HELPERS/game.dm:132]: This doesn't describe what it actually does, which is more important than its implementation.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__HELPERS/game.dm (hunk 4 above)
Comment (Mothblocks) [reply code/__HELPERS/game.dm]: This wasn't resolved.

**Pull request merged**